src/react/databrowser/objects/__tests__/Tags.test.jsx

```diff
@@ -0,0 +1,91 @@
+import * as s3objects from '../../../actions/s3object';
+import { AddButton, Item, SubButton } from '../../../ui-elements/EditableKeyValue';
+import { OBJECT_METADATA } from '../../../actions/__tests__/utils/testUtil';
+import React from 'react';
+import Tags from '../details/Tags';
+import { reduxMount } from '../../../utils/test';
+
+describe('Tags', () => {
+    const putObjectTaggingMock = jest.spyOn(s3objects, 'putObjectTagging').mockReturnValue({ type: '' });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const tests = [
+        'Tags should render',
+        'should render by default an Item with empty values in each input when there are no key/value present',
+        'should add new key/value tag and should trigger function if save button is pressed',
+        'remove button and add button should be disabled',
+    ];
+
+    tests.forEach((t, index) => {
+        it(t, async () => {
+            const { component } = reduxMount(<Tags objectMetadata={OBJECT_METADATA}/>);
+
+            if (index === 0) {
+                expect(component.find(Tags).isEmptyRender()).toBe(false);
+            } else if (index === 1) {
+                expect(component.find('input#tags-input-key').prop('value')).toBe('');
```

Comment:
We can first check that only one "row" is returned before checking its value.

Reply:
done